app/api/events.py

```diff
@@ -121,10 +121,19 @@ def before_post(self, args, kwargs, data=None):
         if data.get('state', None) == 'published' and not is_verified:
             raise ForbiddenException({'source': ''},
                                      "Only verified accounts can publish events")
-        if data.get('state', None) == 'published' and not data.get('location_name', None):
+        if data.get('state', None) == 'published' and not data.get('location_name', None) and \
+                not data.get('is_online'):
             raise ConflictException({'pointer': '/data/attributes/location-name'},
                                     "Location is required to publish the event")
 
+        if data.get('location_name', None) and data.get('is_online'):
+            raise ConflictException({'pointer': '/data/attributes/location-name'},
+                                    "Online Event does not have any locaton")
```

Comment:
I will eventually hide the location fields in orga app in case of online events so you may as well remove this

Reply:
This is not particular to orga-app even a user who is creating event directly using APIs should also mind this thing.